contracts/AelinPool.sol

```diff
@@ -385,6 +443,103 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
+            require(tmpNftData.collectionAddress == _collectionAddress, "collection not in the pool");
+
+            if (tmpNftData.purchaseAmountPerToken) {
+                maxPurchaseTokenAmount += tmpNftData.purchaseAmount * _tokenIds.length;
+            }
+
+            if (!tmpNftData.purchaseAmountPerToken && tmpNftData.purchaseAmount > 0) {
+                require(!nftWalletUsedForPurchase[_collectionAddress][msg.sender], "wallet already used for collection");
+                nftWalletUsedForPurchase[_collectionAddress][msg.sender] = true;
+                maxPurchaseTokenAmount += tmpNftData.purchaseAmount;
+            }
+
+            if (tmpNftData.purchaseAmount == 0) {
+                maxPurchaseTokenAmount = _purchaseTokenAmount;
+            }
+
+            if (NftCheck.supports721(_collectionAddress)) {
+                _purchasePoolWith721(_collectionAddress, _tokenIds);
+            }
+            if (NftCheck.supports1155(_collectionAddress)) {
+                _purchasePoolWith1155(_collectionAddress, _tokenIds, tmpNftData);
+            }
+            if (_collectionAddress == CRYPTO_PUNKS) {
+                _purchasePoolWithPunks(_collectionAddress, _tokenIds);
+            }
+        }
+
+        require(_purchaseTokenAmount <= maxPurchaseTokenAmount, "purchase amount should be less the max allocation");
+
+        uint256 amountBefore = IERC20(purchaseToken).balanceOf(address(this));
+        IERC20(purchaseToken).safeTransferFrom(msg.sender, address(this), _purchaseTokenAmount);
+        uint256 amountAfter = IERC20(purchaseToken).balanceOf(address(this));
+        uint256 purchaseTokenAmount = amountAfter - amountBefore;
+
+        if (purchaseTokenCap > 0) {
+            uint256 totalPoolAfter = totalSupply() + purchaseTokenAmount;
+            require(totalPoolAfter <= purchaseTokenCap, "cap has been exceeded");
+            if (totalPoolAfter == purchaseTokenCap) {
+                purchaseExpiry = block.timestamp;
+            }
+        }
+
+        _mint(msg.sender, purchaseTokenAmount);
+    }
+
+    function _purchasePoolWith721(address _collectionAddress, uint256[] memory _tokenIds) internal {
+        for (uint256 i = 0; i < _tokenIds.length; i++) {
+            require(IERC721(_collectionAddress).ownerOf(_tokenIds[i]) == msg.sender, "has to be the owner of the token");
+            require(!nftId[_collectionAddress][_tokenIds[i]], "tokenId already used");
+            nftId[_collectionAddress][_tokenIds[i]] = true;
+        }
+    }
+
+    function _purchasePoolWith1155(
+        address _collectionAddress,
+        uint256[] memory _tokenIds,
+        NftData memory tmpNftData
+    ) internal view {
+        for (uint256 i = 0; i < _tokenIds.length; i++) {
+            require(nftId[_collectionAddress][_tokenIds[i]], "tokenId not in the pool");
+            require(
+                IERC1155(_collectionAddress).balanceOf(msg.sender, _tokenIds[i]) >= tmpNftData.minTokensEligible[i],
```

Comment:
I think we need to do a check that all the values in minTokensEligible are greater than or equal to 1. They can't be set to 0 basically

Reply:
i think we can let that depend on the `sponsor` else we need to check `tokenIds.length == minTokensEligible.length`  also